impeller/display_list/canvas.cc

```diff
@@ -1066,14 +1075,40 @@ void Canvas::SaveLayer(const Paint& paint,
           return filter;
         };
 
-    auto input_texture = FlipBackdrop(render_passes_,           //
-                                      GetGlobalPassPosition(),  //
-                                      clip_coverage_stack_,     //
-                                      renderer_                 //
-    );
-    if (!input_texture) {
-      // Validation failures are logged in FlipBackdrop.
-      return;
+    std::shared_ptr<Texture> input_texture;
+
+    // If the backdrop ID is not the no-op id, and there is more than one usage
+    // of it in the current scene, cache the backdrop texture and remove it from
+    // the current entity pass flip.
+    bool will_cache_backdrop_texture = false;
+    BackdropData* backdrop_data = nullptr;
+    if (backdrop_id.has_value()) {
+      const auto& backdrop_data_it = backdrop_data_.find(backdrop_id.value());
```

Comment:
```suggestion
      std::unordered_map<int64_t, BackdropData>::iterator backdrop_data_it = backdrop_data_.find(backdrop_id.value());
```
We're grabbing a reference to an rvalue.

Reply:
Done

Reply:
(You didn't push the commit yet)

Reply:
Okay now I did :)